### DISCUSSION TITLE
Why are html nodes are not merged for some tags?

### DISCUSSION THREAD

**Discussion by brc-dd** [remarkjs/remark#1387] [Q&A]
Currently some code like this

```ts
import { unified } from 'unified'
import remarkParse from 'remark-parse'

const processor = unified().use(remarkParse)

console.log(processor.parse('<template></template>'))
console.log(processor.parse('<div></div>'))
```

logs

```jsonc
{
  type: "root",
  children: [
    {
      type: "paragraph",
      children: [
        { type: "html", value: "<template>", position: [Object] },
        { type: "html", value: "</template>", position: [Object] }
      ],
      position: {
        start: { line: 1, column: 1, offset: 0 },
        end: { line: 1, column: 22, offset: 21 }
      }
    }
  ],
  position: {
    start: { line: 1, column: 1, offset: 0 },
    end: { line: 1, column: 22, offset: 21 }
  }
}
{
  type: "root",
  children: [
    {
      type: "html",
      value: "<div></div>",
      position: {
        start: { line: 1, column: 1, offset: 0 },
        end: { line: 1, column: 12, offset: 11 }
      }
    }
  ],
  position: {
    start: { line: 1, column: 1, offset: 0 },
    end: { line: 1, column: 12, offset: 11 }
  }
}
```

Why is there difference in the two cases? 👀 

---

Edit: Ah, I think it's because template is not recognized as a html tag. Is there some way to configure/extend the parser so that template or any other custom tag is treated same as normal html elements?

---

I see that there was a blocks property in v8 but removed in v9.

**Comment by wooorm** [comment]
It has to do with how markdown works. There are several patterns that match "block" html. Which is something that works per line and that is emitted verbatim. Otherwise, inside paragraphs (and headings and such) there is "inline" html. That only matches separate tags and such. So that you can embed some more markdown (like emphasis) inside those tags.
